app/src/main/java/fr/free/nrw/commons/CommonsAppAdapter.java

```diff
@@ -0,0 +1,72 @@
+package fr.free.nrw.commons;
+
+import org.wikipedia.AppAdapter;
+import org.wikipedia.dataclient.Service;
+import org.wikipedia.dataclient.SharedPreferenceCookieManager;
+import org.wikipedia.dataclient.WikiSite;
+import org.wikipedia.login.LoginResult;
+
+import javax.inject.Inject;
+
+import androidx.annotation.NonNull;
+import fr.free.nrw.commons.auth.SessionManager;
+import okhttp3.OkHttpClient;
+
+public class CommonsAppAdapter extends AppAdapter {
+    private final int DEFAULT_THUMB_SIZE = 640;
+
+    @Inject SessionManager sessionManager;
```

Comment:
AFAIK `@Inject` annotation can be used along with something like `AndroidInjection.inject(context)`. `context` is directly available only in android base classes. 

Another way to inject is to have a constructor annotated with `@Inject`. 

To make things clear I opened a PR to demonstrate the usage: https://github.com/dbrant/apps-android-commons/pull/1

Reply:
Not sure I'm understanding this. The only reason I used `@Inject` here is because that's how it is used everywhere else.  The `CommonsAppAdapter` class itself won't need to be injected anywhere.

Reply:
It is used this way in activity/fragment classes for eg. `UploadActivity`. 

And for other classes, the constructor is annotated with `@Inject`. eg. `UploadModel`